### PR TITLE
Tracking Plasma Token Mappings

### PR DESCRIPTION
**Pull request by itzmeanjan**
## changes

This PR attempts to make Plasma & POS token mapping tracking easier while distinguishing them, using binary value.

### CODE REVIEW

**Review by Meshugah — APPROVED** [review]
The registry contract on goerli has a couple of reverted transactions. Just pointing it out: https://goerli.etherscan.io/tx/0x936576973320be3057ea8de858efc0baed2fb1e96dbd8bd09146e837ebc0701a.

**Review by reddyismav — APPROVED** [review]

**Comment by itzmeanjan** [comment]
> The registry contract on goerli has a couple of reverted transactions. Just pointing it out: https://goerli.etherscan.io/tx/0x936576973320be3057ea8de858efc0baed2fb1e96dbd8bd09146e837ebc0701a.

Doesn't seem those concern us.

**Review by reddyismav — APPROVED** [review]

File: root/src/mappings/registry.ts [hunk 1]
```diff
@@ -0,0 +1,21 @@
+import { TokenMapped } from '../../generated/Registry/Registry'
+import { TokenMapping } from '../../generated/schema'
+
+export function handlePlasmaTokenMapped(event: TokenMapped): void {
+  let id = 'token-mapping-' + event.params.rootToken.toHexString()
```
Comment (jdkanani) [inline root/src/mappings/registry.ts:5]: What happens when we have token mapped on both plasma and pos?

We should add test in id:

```suggestion
  let id = 'plasma-token-mapping-' + event.params.rootToken.toHexString()
```

File: root/subgraph.template.yaml [hunk 2]
```diff
@@ -73,8 +73,29 @@ dataSources:
...
+    source:
+      address: "{{contracts.registry.address}}"
+      abi: Registry
+      startBlock: {{ contracts.registry.startBlock }}
+    mapping:
+      kind: ethereum/events
+      apiVersion: 0.0.4
+      language: wasm/assemblyscript
+      entities:
+        - TokenMapped
```
Comment (jdkanani) [inline root/subgraph.template.yaml:91]:
```suggestion
        - TokenMapping
```

**Review by jdkanani — CHANGES_REQUESTED** [review]

File: root/subgraph.template.yaml (hunk 2 above)
Comment (itzmeanjan) [reply root/subgraph.template.yaml]: Fixed

File: root/src/mappings/registry.ts (hunk 1 above)
Comment (itzmeanjan) [reply root/src/mappings/registry.ts]: Fixed issue, using different `id` for Plasma & PoS root tokens.

Comment (itzmeanjan) [reply root/src/mappings/registry.ts]: Thanks for pointing it out @jdkanani 🙌

**Review by jdkanani — APPROVED** [review]

**Pull request merged**